### ISSUE TITLE
Conflict between microphone and speaker on the IOS device.

### ISSUE THREAD

**Issue by igor-nm** [ak1394/react-native-tts#61]
In IOS when permission is granted to the microphone, the device's audio goes to the upper speaker and the sound becomes low because that speaker serves for the purpose of call. Please, anybody can help me?

**Comment by rodrigosol** [comment]
Same problem here!

**Comment by dalinarkholin** [comment]
@ak1394 @wenkesj I am seeing this as well. When playing audio, and simply running `Voice.start()` , the phone ear speaker is used instead of the body speakers. I don't think this was happening a couple weeks ago. I think there might have been a ios change. Any ideas?

I am running IOS 11.2.5

Not sure where we want to keep this thread. Also have a comment here: https://github.com/wenkesj/react-native-voice/issues/75

My gut is telling me this is a intentional change by ios. Maybe something to do with the new HomePod support they built in. https://support.apple.com/en-us/HT208067

For my case, I don't actually need both the audio and voice recognition to be happening at the sametime. If we introduced promises(await async to this module that might solve the issue. That way I can wait on `tts.speak(someString)` to finish before enabling the microphone via `Voice.start()`.

**Comment by ak1394** [comment]
Guys, I'm not sure I can be much of help here, this seems like IOS issue? As of returning promises from speak(), for now you should be able to use events to wait for it to finish speaking. I have to document it but speak() returns an utteranceId and if you subscribe with `Tts.addEventListener('tts-finish', (event) => console.log("finish", event)); ` you'll see the same id in the event as well.

**Comment by dalinarkholin** [comment]
Thanks @ak1394. That was helpful! Moving this conversation back over to react-native-voice for now.

**Comment by igor-nm** [comment]
Hi guys, everything fine?
I would like to thank you guys for looking at this problem, do you guys think we can find a solution to this, or is it really an iOS problem?
I do ever not developing in Objective-C so I can not help them much with this.
Sorry me. :(

**Comment by igor-nm** [comment]
This problem was resolved that topic: https://github.com/wenkesj/react-native-voice/issues/75
The solution was found pelo @dalinarkholin, and I already asked him if can release pull request.
Thank's everyone.

**Comment by jferas** [comment]
Were a pull requests ever done for tts and voice related to this issue?  Looking at the commit logs I don't see it.  If not, I'll do it, because the fork belonging to @dalinarkholin is not rn 0.56 compatible and I want to get this fix into an app that I'm working on.  This issue caught be by surprise because I was testing in iPad and didn't see it (no ear speaker) and gave the app to a friend and he noticed on his iPhone.

**Comment by dalinarkholin** [comment]
@jferas It was never merged. I didn't feel comfortable with the solution and Im pretty sure it could be resolved just on the react-native-voice package.  You can see my solution in my commits here:
 [https://github.com/dalinarkholin/react-native-voice/commits/master](https://github.com/dalinarkholin/react-native-voice/commits/master)

**Comment by jferas** [comment]
@dalinarkholin Thanks!.. I saw your commits in react-native-voice and re-did them in a fork/branch from master that had the react native 0.56 support in it.  Seems to be working fine in my app, so I did a pull request for it over in react-native-voice.  Thanks for doing original legwork!  Strange that we're posting about it here in react-native-tts but this is where the problem surfaced, and of course where I first went when I noticed it.
